File: nodejs/source/dexi/DexiBinaryResponse.js
/**
 * Wraps a HTTP response which contains binary data
 * @param response {DexiAPIResponse}
 * @constructor
 */
function DexiBinaryResponse(response) {
    /**
     *
     * @type {string}
     */
    this.data = response.getResponseBody();

    /**
     *
     * @type {string}
     */
    this.mimeType = response.getHeaders()['Content-Type'];
}

/**
 *
 * @returns {string}
 */
DexiBinaryResponse.prototype.getData = function() {
    return this.data;
};

/**
 *
 * @returns {string}
 */
DexiBinaryResponse.prototype.getMimeType = function() {
    return this.mimeType;
};

module.exports = DexiBinaryResponse;

File: java/source/src/main/java/com/dexi/client/DexiBinaryResponse.java
package com.dexi.client;

public class DexiBinaryResponse  {
    private final byte[] data;
    private final String mimeType;

    public DexiBinaryResponse (DexiAPIResponse response) {
        this.data = response.getResponseBody();
        this.mimeType = response.getHeaders().get("Content-Type");
    }

    public byte[] getData() {
        return data;
    }

    public String getMimeType() {
        return mimeType;
    }
}
